src/app/pages/fund/choose-asset-to-fund/choose-asset-to-fund.tsx

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useMemo } from 'react';
+import { createSearchParams, useNavigate } from 'react-router-dom';
+
+import { AllTransferableCryptoAssetBalances } from '@shared/models/crypto-asset-balance.model';
+import { RouteUrls } from '@shared/route-urls';
+
+import { useStxBalance } from '@app/common/hooks/balance/stx/use-stx-balance';
+import { useRouteHeader } from '@app/common/hooks/use-route-header';
+import { useWalletType } from '@app/common/use-wallet-type';
+import { ChooseAssetContainer } from '@app/components/crypto-assets/choose-crypto-asset/choose-asset-container';
+import { ChooseCryptoAssetLayout } from '@app/components/crypto-assets/choose-crypto-asset/choose-crypto-asset.layout';
+import { CryptoAssetList } from '@app/components/crypto-assets/choose-crypto-asset/crypto-asset-list';
+import { ModalHeader } from '@app/components/modal-header';
+import { useNativeSegwitBalance } from '@app/query/bitcoin/balance/btc-native-segwit-balance.hooks';
+import { createStacksCryptoCurrencyAssetTypeWrapper } from '@app/query/stacks/balance/stacks-ft-balances.utils';
+import { ModalBackgroundWrapper } from '@app/routes/components/modal-background-wrapper';
+import { useHasCurrentBitcoinAccount } from '@app/store/accounts/blockchain/bitcoin/bitcoin.hooks';
+import { useCurrentAccountNativeSegwitSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useHasStacksLedgerKeychain } from '@app/store/accounts/blockchain/stacks/stacks.hooks';
+
+function useBtcCryptoCurrencyAssetBalance() {
+  const currentBtcSigner = useCurrentAccountNativeSegwitSigner();
+  if (!currentBtcSigner?.(0).address) throw new Error('No bitcoin address');
+
+  return useNativeSegwitBalance(currentBtcSigner?.(0).address);
+}
```

Comment:
We have a hook to handle all balance related data reads? Why does the fund page need to implement a new hook? See `useBtcAssetBalance`

We aim to move away from native segwit being your balance, towards any tr or ns funds being your balance. In duplicating the logic here, we'll make change harder later on.

Reply:
Good point! Do I understand correctly that i would still need this custom hook tho? As I need to get an address from the signer that i later on pass to `useBtcAssetBalance` hook.
```suggestion
function useBtcCryptoCurrencyAssetBalance() {
  const currentBtcSigner = useCurrentAccountNativeSegwitSigner();
  if (!currentBtcSigner?.(0).address) throw new Error('No bitcoin address');

  return useBtcAssetBalance(currentBtcSigner?.(0).address).btcAssetBalance;
}
```

Reply:
This is one way of doing it. In some places, we've started using components to pass in the data. This can be preferable, as then you don't need to use a hook that might be undefined. See `CurrentBitcoinAccountLoader`.